### ISSUE TITLE
Is reclustering possible?

### ISSUE THREAD

**Issue by sv-bio** [spyking-circus/spyking-circus#343]
Dear Pierre, 

When I visualize results in Phy2 rather often I can see that some clusters are not homogenous (consist of spikes of somewhat different shapes). Is it possible to select these clusters in Phy2 and run Spyking Circus on them once again (recluster them) to break them into more homogenous sub-clusters and hense improve separation?

Sincerely,
Stanislav

**Comment by sv-bio** [comment]
I wrote a plug-in for Phy2, that does K-means re-clustering on selected clusters. Am new at Python, so the plugin is mainly copy-paste from the other sources (thanks to Peter C. Petersen (https://github.com/petersenpeter/phy2-plugins/blob/master/plugins/recluster.py) and Cyrille Rossant (Phy2's creator). I think, issue can be closed.

Plugin:

from phy import IPlugin, connect
import numpy as np
import logging

logger = logging.getLogger('phy')

def k_means(x):
    from sklearn.cluster import KMeans
    return KMeans(n_clusters=5).fit_predict(x)

class CustomSplitPlugin(IPlugin):
    def attach_to_controller(self, controller):
        @connect
        def on_gui_ready(sender, gui):
            @controller.supervisor.actions.add(shortcut='s')

            def custom_split():

                cluster_ids = controller.supervisor.selected
                spike_ids=controller.supervisor.clustering.spikes_in_clusters(cluster_ids)

                data = controller.model._load_features()
                data3 = data.data[spike_ids]
                data2 = np.reshape(data3,(data3.shape[0],data3.shape[1]*data3.shape[2]))

                labels = k_means(data2)
                assert spike_ids.shape == labels.shape
                controller.supervisor.actions.split(spike_ids, labels)
